From: Michael Fladischer <FladischerMichael@fladi.at>
Date: Wed, 28 Mar 2018 21:08:22 +0200
Subject: Disable pytest (>=3.3) log capturing to avoid changing log handlers
 (Closes: #892640).

---
 setup.cfg | 1 +
 1 file changed, 1 insertion(+)

diff --git a/setup.cfg b/setup.cfg
index 09f05f8..b056c41 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -1,6 +1,7 @@
 [tool:pytest]
 testpaths = t/unit/
 python_classes = test_*
+addopts=-p no:logging
 
 [build_sphinx]
 source-dir = docs/
