--- celery.orig/celery/apps/multi.py
+++ celery/celery/apps/multi.py
@@ -198,7 +198,7 @@ class Node:
             try:
                 os.kill(pid, sig)
             except OSError as exc:
-                if exc.errno != errno.ESRCH:
+                if (exc.errno != errno.ESRCH and exc.errno != errno.EPERM):
                     raise
                 maybe_call(on_error, self)
                 return False
