diff --git a/extra/generic-init.d/celeryd b/extra/generic-init.d/celeryd
index 0fe704f..ae7eab9 100755
--- a/extra/generic-init.d/celeryd
+++ b/extra/generic-init.d/celeryd
@@ -232,12 +232,22 @@ start_workers () {
     if [ ! -z "$CELERYD_ULIMIT" ]; then
         ulimit $CELERYD_ULIMIT
     fi
-    _chuid $* start $CELERYD_NODES $DAEMON_OPTS     \
-                 --pidfile="$CELERYD_PID_FILE"      \
-                 --logfile="$CELERYD_LOG_FILE"      \
-                 --loglevel="$CELERYD_LOG_LEVEL"    \
-                 $CELERY_APP_ARG                    \
-                 $CELERYD_OPTS
+
+    start-stop-daemon --start                   \
+        --chuid $CELERYD_USER                   \
+        --user $CELERYD_USER                    \
+        --group $CELERYD_USER                   \
+        --quiet                                 \
+        --pidfile $CELERYD_PRIMARY_PID_FILE     \
+        --oknodo                                \
+        --chdir $CELERYD_CHDIR                  \
+        --exec $CELERYD_MULTI $* -- start       \
+            $CELERYD_NODES $DAEMON_OPTS         \
+            --pidfile="$CELERYD_PID_FILE"       \
+            --logfile="$CELERYD_LOG_FILE"       \
+            --loglevel="$CELERYD_LOG_LEVEL"     \
+            $CELERY_APP_ARG                     \
+            $CELERYD_OPTS
 }
 
 
@@ -247,22 +257,18 @@ dryrun () {
 
 
 stop_workers () {
-    _chuid stopwait $CELERYD_NODES --pidfile="$CELERYD_PID_FILE"
+    $CELERYD_MULTI $* stopwait $CELERYD_NODES --pidfile="$CELERYD_PID_FILE"
 }
 
 
 restart_workers () {
-    _chuid restart $CELERYD_NODES $DAEMON_OPTS      \
-                   --pidfile="$CELERYD_PID_FILE"    \
-                   --logfile="$CELERYD_LOG_FILE"    \
-                   --loglevel="$CELERYD_LOG_LEVEL"  \
-                   $CELERY_APP_ARG                  \
-                   $CELERYD_OPTS
+    stop_workers
+    start_workers
 }
 
 
-kill_workers() {
-    _chuid kill $CELERYD_NODES --pidfile="$CELERYD_PID_FILE"
+kill_workers () {
+    $CELERYD_MULTI $* kill $CELERYD_NODES --pidfile="$CELERYD_PID_FILE"
 }
 
 
